beacon-chain/db/filesystem/data_column.go

```diff
@@ -0,0 +1,1014 @@
+package filesystem
+
+import (
+	"context"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/OffchainLabs/prysm/v6/async"
+	"github.com/OffchainLabs/prysm/v6/async/event"
+	"github.com/OffchainLabs/prysm/v6/beacon-chain/verification"
+	fieldparams "github.com/OffchainLabs/prysm/v6/config/fieldparams"
+	"github.com/OffchainLabs/prysm/v6/config/params"
+	"github.com/OffchainLabs/prysm/v6/consensus-types/blocks"
+	"github.com/OffchainLabs/prysm/v6/consensus-types/primitives"
+	"github.com/OffchainLabs/prysm/v6/io/file"
+	"github.com/OffchainLabs/prysm/v6/time/slots"
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+)
+
+const (
+	version                            = 0x01
+	versionOffset                      = 0                           // bytes
+	versionSize                        = 1                           // bytes
+	maxSszEncodedDataColumnSidecarSize = 536_870_912                 // 2**(4*8) / 8 bytes
+	sidecarSizeOffset                  = versionOffset + versionSize // (Offset of the encoded size of the SSZ encoded data column sidecar)
+	sidecarSizeSize                    = 4                           // bytes (Size of the encoded size of the SSZ encoded data column sidecar)
+	mandatoryNumberOfColumns           = 128                         // 2**7
+	indicesOffset                      = sidecarSizeOffset + sidecarSizeSize
+	indicesSize                        = mandatoryNumberOfColumns
```

Comment:
not used

Reply:
Fixed in 14f0ab638e048e907c426f008471ecd03c306cc2.